### PR TITLE
patch GKE machine deployment version and size

### LINKED ISSUE #8249: patch GKE machine deployment version and size
patch GKE machine deployment version and size

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: patch GKE machine deployment version and size

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #8249

```release-note
NONE
```

### CODE REVIEW

**Comment by zreigz** [comment]
@imharshita PTAL

File: pkg/handler/v2/external_cluster/gke.go [hunk 1]
```diff
@@ -192,3 +209,65 @@ func deleteGKENodePool(ctx context.Context, cluster *kubermaticapiv1.ExternalClu
...
+	// only when size was updates otherwise change NodePoll object with other parameters
+	if old.Spec.Replicas != new.Spec.Replicas {
+		sizeRequest := &container.SetNodePoolSizeRequest{
+			NodeCount: int64(new.Spec.Replicas),
+		}
+		sizeReq := svc.Projects.Zones.Clusters.NodePools.SetSize(project, cluster.Spec.CloudSpec.GKE.Zone, cluster.Spec.CloudSpec.GKE.Name, old.Name, sizeRequest)
+		_, err = sizeReq.Context(ctx).Do()
+		if err != nil {
+			return nil, err
+		}
```
Comment (imharshita) [inline pkg/handler/v2/external_cluster/gke.go:235]: Can you before returning new here update this as new has both replica and version patch `new.NodeDeployment.Spec.Template.Versions.Kubelet = old.Spec.Template.Versions.Kubelet`

**Review by imharshita — CHANGES_REQUESTED** [review]

File: pkg/handler/v2/external_cluster/gke.go [hunk 2]
```diff
@@ -192,3 +209,65 @@ func deleteGKENodePool(ctx context.Context, cluster *kubermaticapiv1.ExternalClu
 	_, err = req.Context(ctx).Do()
 	return err
 }
+
+func patchGKEMachineDeployment(ctx context.Context, old, new *apiv2.ExternalClusterMachineDeployment, cluster *kubermaticapiv1.ExternalCluster, secretKeySelector provider.SecretKeySelectorValueFunc, credentialsReference *providerconfig.GlobalSecretKeySelector, clusterProvider provider.ExternalClusterProvider) (*apiv2.ExternalClusterMachineDeployment, error) {
```
Comment (imharshita) [inline pkg/handler/v2/external_cluster/gke.go]: clusterProvider not used here

Comment (zreigz) [reply pkg/handler/v2/external_cluster/gke.go]: I will fix this in the next PR

**Pull request merged**